tests/stub/routing/_routing.py

```diff
@@ -0,0 +1,154 @@
+from abc import abstractmethod
+from collections import defaultdict
+try:
+    import fcntl
+except ImportError:  # e.g. on Windows
+    fcntl = None
+import inspect
+import json
+import os
+import socket
+import struct
+from sys import platform
+
+from nutkit.frontend import Driver
+import nutkit.protocol as types
+from tests.shared import (
+    get_driver_name,
+    TestkitTestCase,
+)
+from tests.stub.shared import StubServer
+
+
+def get_extra_hello_props():
+    if get_driver_name() in ["java"]:
+        return ', "realm": ""'
+    elif get_driver_name() in ["javascript"]:
+        return ', "realm": "", "ticket": ""'
+    return ""
+
+
+# This should be the latest/current version of the protocol.
+# Older protocol that needs to be tested inherits from this and override
+# to handle variations.
+class RoutingBase(TestkitTestCase):
```

Comment:
Maybe it could be called `RoutingCurrent` or `RoutingLatest` since the description is `should be the latest/current version of the protocol`

Reply:
I was thinking about it :thinking: